Frends.Web/Web.cs

```diff
@@ -651,7 +700,59 @@ internal static void SetDefaultRequestHeadersBasedOnOptions(this HttpClient http
             httpClient.Timeout = TimeSpan.FromSeconds(Convert.ToDouble(options.ConnectionTimeoutSeconds));
         }
 
-        internal static X509Certificate2 GetCertificate(string thumbprint)
+        private static X509Certificate[] GetCertificates(Options options)
+        {
+            X509Certificate2[] certificates;
+
+            switch (options.ClientCertificateSource)
+            {
+                case CertificateSource.CertificateStore:
+                    var thumbprint = options.CertificateThumbprint;
+                    certificates = GetCertificatesFromStore(thumbprint, options.LoadEntireChainForCertificate);
+                    break;
+                case CertificateSource.File:
+                    certificates = GetCertificatesFromFile(options.ClientCertificateFilePath, options.ClientCertificateKeyPhrase);
+                    break;
+                case CertificateSource.String:
+                    certificates = GetCertificatesFromString(options.ClientCertificateInBase64, options.ClientCertificateKeyPhrase);
+                    break;
+                default:
+                    throw new Exception("Unsupported Certificate source");
+            }
+
+            return certificates.Cast<X509Certificate>().ToArray();
+        }
+
+        private static X509Certificate2[] GetCertificatesFromString(string certificateContentsBase64, string keyPhrase)
+        {
+            var certificateBytes = Convert.FromBase64String(certificateContentsBase64);
+
+            return LoadCertificatesFromBytes(certificateBytes, keyPhrase);
+        }
+
+        private static X509Certificate2[] LoadCertificatesFromBytes(byte[] certificateBytes, string keyPhrase)
+        {
+            var collection = new X509Certificate2Collection();
+
+            if (!string.IsNullOrEmpty(keyPhrase))
+            {
+                collection.Import(certificateBytes, keyPhrase, X509KeyStorageFlags.PersistKeySet);
+            }
+            else
+            {
+                collection.Import(certificateBytes, null, X509KeyStorageFlags.PersistKeySet);
+            }
+            return collection.Cast<X509Certificate2>().OrderByDescending(c => c.HasPrivateKey).ToArray();
+
+        }
+
+        private static X509Certificate2[] GetCertificatesFromFile(string clientCertificateFilePath, string keyPhrase)
+        {
+            return LoadCertificatesFromBytes(File.ReadAllBytes(clientCertificateFilePath), keyPhrase);
+        }
+
+        private static X509Certificate2[] GetCertificatesFromStore(string thumbprint,
+            bool loadEntireChain)
         {
             thumbprint = Regex.Replace(thumbprint, @"[^\da-zA-z]", string.Empty).ToUpper();
             var store = new X509Store(StoreName.My, StoreLocation.CurrentUser);
```

Comment:
The X509Store is disposable from .net framework 4.6 => 
This could be changed to using instead of finally close() ?